ringpop.go

```diff
@@ -89,8 +90,9 @@ type Ringpop struct {
 		sync.RWMutex
 	}
 
-	logger log.Logger
-	log    log.Logger
+	logFacility logger.LogFacility
```

Comment:
I think this is a bit of a smell. With the logging work, it's not any clearer why (at a glance) there is the distinction between `logger` and `log`, but now we have a 3rd nebulous concept in `logFacilty`. What's the distinction between all 3 of these? Do we need all of them?